packages/framework/ai-collab/src/test/explicit-strategy/utils.ts

```diff
@@ -76,3 +78,101 @@ export function initializeOpenAIClient(service: "openai" | "azure"): OpenAI {
 		return client;
 	}
 }
+
+/**
+ * A utility class for snapshot testing.
+ */
+export class MochaSnapshotUnitTester {
+	public constructor(
+		public readonly snapshotDirectory: string,
+		public readonly suiteName: string,
+	) {}
+
+	public expectToMatchSnapshot(
+		mochaContext: Mocha.Context,
+		output: string,
+		options?: { fileNameOverride?: string; metadata?: Record<string, string> },
```

Comment:
I think I always saw `fileNameOverride` being applied... does it make sense to just make it a required parameter (and name it `snapshotFilename`)? I think it'll also help for when a given test is renamed. If no override is specified in that case, we'll get a new snapshot file and nothing will clean the old one. If we at least keep the string used for the filename, the test can be renamed but we guarantee that the filename will remain the same unless the developer explicitly changes it too and thinks about the implications.

Reply:
I was going back and forth on this, because right now it gives you an automatic name using the tests name when an override is not provided, so its a convenience thing.

I found myself preferring to use file name overrides for reasons you mentioned -- legibility & not tying it to the actual test name. I think it makes sense to make the file name required because otherwise there is no way to easily search and find the connection between a given snapshot and the test code that's running against/producing it also.

Updated in commit 6fc87d9de1a64f7fb50d2294ce8a29b8ced9e904